exercises/bank-account/BankAccountSpec.groovy

```diff
@@ -0,0 +1,202 @@
+        when:
+            bankAccount.open();
+            bankAccount.deposit(-1);
+
+        then:
+            thrown(Exception)                   
+    }
+
+    def cannotWithdrawNegativeAmount() {
+        setup:
+            BankAccount bankAccount=new BankAccount()
+        when:
+            bankAccount.open();
+            bankAccount.deposit(105);
+            bankAccount.withdraw(-5);
+
+        then:
+            thrown(Exception)                   
+    }
+
+    def cannotGetBalanceOfClosedAccount() {
+        setup:
+            BankAccount bankAccount=new BankAccount()
+        when:
+            bankAccount.open();
+            bankAccount.deposit(10);
+            bankAccount.close();
+            bankAccount.getBalance();
+
+        then:
+            thrown(Exception)                   
+    }
+
+    def cannotDepositMoneyIntoClosedAccount() {
+        setup:
+            BankAccount bankAccount=new BankAccount()
+        when:
+            bankAccount.open();
+            bankAccount.close();
+            bankAccount.deposit(5);
+
+        then:
+            thrown(Exception)                   
+    }
+
+    def cannotWithdrawMoneyFromClosedAccount() {
+        setup:
+            BankAccount bankAccount=new BankAccount()
+        when:
+            bankAccount.open();
+            bankAccount.deposit(20);
+            bankAccount.close();
+            bankAccount.withdraw(5);
+
+        then:
+            thrown(Exception)                   
+    }
+
+    def bankAccountIsClosedBeforeItIsOpened() {
+        setup:
+            BankAccount bankAccount=new BankAccount()
+        when:
+            bankAccount.getBalance();
+
+        then:
+            thrown(Exception)                   
+    }
+
+    def canAdjustBalanceConcurrently() {
+        setup:
+            BankAccount bankAccount=new BankAccount();
+        when:
+            bankAccount.open();
+            bankAccount.deposit(1000);
+            for (int i = 0; i < 10; i++) {
+                adjustBalanceConcurrently(bankAccount);
+            }
+        
+        then:
+            bankAccount.getBalance() == 1000
+    }
+
+    void adjustBalanceConcurrently(BankAccount bankAccount) {
+        Random random = new Random();
+
+        Thread[] threads = new Thread[1000];
+        for (int i = 0; i < 1000; i++) {
+            threads[i] = new Thread(
+                { 
+                        try {
+                            bankAccount.deposit(5);
+                            Thread.sleep(random.nextInt(10));
+                            bankAccount.withdraw(5);
+                        } catch (InterruptedException ignored) {
+                        } catch (Exception e) {
+                            fail("Exception should not be thrown: " + e.getMessage());
+                        } 
+                }
+            );
+            threads[i].start();
+        }
+
+        for (Thread thread : threads) {
```

Comment:
the same thing:
```groovy
l.each { println it }
```
These two comments are just for your considerations and it isn't mandatory to change anything, but what's the point of using Groovy just by copying Java code, right?